src/swarm-tui/Swarm/TUI/Controller.hs

```diff
@@ -693,7 +693,12 @@ handleREPLEventTyping = \case
         uiState . uiGameplay . uiREPL . replPromptType %= \case
           CmdPrompt _ -> CmdPrompt [] -- reset completions on any event passed to editor
           SearchPrompt a -> SearchPrompt a
-        modify validateREPLForm
+
+        -- Now re-validate the input, unless only the cursor moved.
+        case ev of
+          Key V.KLeft -> pure ()
+          Key V.KRight -> pure ()
+          _ -> modify validateREPLForm
```

Comment:
Note, it would be shorter to write something like 
```
unless (ev `elem` [Key V.KLeft, Key V.KRight]) $ modify validateREPLForm
```
However, that doesn't work, since it requires an `Eq` instance for `Event`, which in turn requires an `Eq` instance for `AppEvent`, but `AppEvent` can contain a function so that's a no-go.

Reply:
Not exactly "better", but:
```
unless (null [() | Key x <- [ev], x `notElem` [V.KLeft, V.KRight]]) $
    modify validateREPLForm
```

Incidentally, I also tried using a refutable pattern with `do` notation and the `Maybe` instance of `MonadFail`, but I couldn't get it to work:

```
let maybeExempt = do
       Key x <- ev
       guard $ x `notElem` [V.KLeft, V.KRight]
       return ()
unless (null maybeExempt) $
    modify validateREPLForm
```
It yields the error:
```
    • Couldn't match expected type ‘AppEvent’
                  with actual type ‘BrickEvent n0 e0’
    • In the pattern: Key x
```

Reply:
Hmm, yes, I hadn't considered using a pattern match in a list comprehension.  Maybe if we add more keys to the list in the future we could consider shortening it like this.  But in this case I think there's virtue in keeping it simple and obvious and not trying to be too clever, even if it means the code is a bit longer.

> I also tried using a refutable pattern with `do` notation and the `Maybe` instance of `MonadFail`, but I couldn't get it to work:

It looks like the types `n0` and `e0` are ambiguous, so it can't reduce `BrickEvent n0 e0` to `AppEvent`.  It could probably be solved by adding some type signatures or explicit type applications, but I'm not motivated to figure out exactly how!